pkg/syncutil/opadataclient.go

```diff
@@ -19,51 +19,20 @@ import (
 	"context"
 
 	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
-	"github.com/open-policy-agent/gatekeeper/v3/pkg/watch"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+// CacheManagerMediator is an interface for mediating
+// with a CacheManager but not actually depending on an instance.
+type CacheManagerMediator interface {
+	AddObject(ctx context.Context, instance *unstructured.Unstructured) error
+	RemoveObject(ctx context.Context, instance *unstructured.Unstructured) error
+
+	ReportSyncMetrics()
+}
+
```

Comment:
the `CacheManager` uses it to interface with the opa client. I don't know why it was added originally but my guesstimate is that it was trying to restrict the API set available to it to the two funcs.

Do you want me to remove it and pass the full opa client to the CacheManager? Does the CacheManager care about all the other funcs in the opa client?

Reply:
If nothing depends on something and we don't know of something that will, I think it's generally best to remove that thing.  Should we need it, we can always go find it in the commit history so we don't have to re-engineer it.

Reply:
> If nothing depends on something and we don't know of something that will, I think it's generally best to remove that thing.

Agree in this instance. One thing to be aware of is that some options are consumed down stream and that's not always obvious (e.g. the ability to turn off referential constraint templates).

Also, generally an interface should be defined by the consumer, so if CacheManager wants to use this interface (can be useful for things like simplifying writing tests -- less to mock), it should be defined in CacheManager's package.

Reply:
Also, since we are actively writing CacheManager... I sincerely hope we're not already running into Chesterton's fence for it :p

Reply:
Could we either 1) keep the OpaDataClient interface around since we also use it for a `FakeOpa` backend for storing data or, if that's not ok, 2) I can commit to remove this and rework the `FakeOpa` in a subsequent PR?

Reply:
Chatted with Alex, it seems like there is value in being able to mock the opa cache when testing the CacheManager (assuming we are testing at that level).  It allows one to make a simpler map and assert its contents.

I agree with Max's assertion that the caller (not the dependency) should define the interface it needs to function correctly.  Based on that, the interface seems to have a home alongside the CacheManager.

If that testing value were not to materialize or decrease over time, it seems sensible to remove the interface at that point.

Reply:
I've moved the `OpaDataClient` interface in the `cachemanager` package.